### PR TITLE
fix(Core/Common): -Wdeprecated-enum-float-conversion warning

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
- Fixes tons of warnings like:

```
src/server/scripts/EasternKingdoms/zone_undercity.cpp:1567:48: fatal error: arithmetic between floating-point type 'double' and enumeration type 'TimeConstants' is deprecated [-Wdeprecated-enum-float-conversion]
                            JumpToNextStep(1.5 * IN_MILLISECONDS);
                                           ~~~ ^ ~~~~~~~~~~~~~~~
```

### CODE REVIEW

**Comment by Kitzunu** [comment]
std::chrono soonTM

File: src/common/Common.h [hunk 1]
```diff
@@ -42,16 +42,14 @@
...
-    IN_MILLISECONDS = 1000
-};
+// TimeConstants
+#define MINUTE 60
+#define HOUR MINUTE * 60
+#define DAY HOUR * 24
+#define WEEK DAY * 7
+#define MONTH DAY * 30
+#define YEAR MONTH * 12
+#define IN_MILLISECONDS 1000
```
Comment (Kitzunu) [inline src/common/Common.h:52]: Isnt it possible to do 
```cpp
enum TimeConstants : uint32
```

Comment (Kitzunu) [reply src/common/Common.h:52]: Maybe not with enum I cant remember, but it is possible to do with enum class. Perhaps that avoids the error and lets us avoid #define

Comment (FrancescoBorzi) [reply src/common/Common.h:52]: I've tried that, still got:

```
Arithmetic between floating-point type 'double' and enumeration type 'TimeConstants' is deprecated
```

when multiplying with decimal values

Comment (Kitzunu) [reply src/common/Common.h:52]: hm.. Good enough I guess until we switch to std::chrono

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**